src/test.spec.ts

```diff
@@ -0,0 +1,124 @@
+import axios from 'axios';
+import { AxiosHarTracker } from './axios-har-tracker'
+import MockAdapter from 'axios-mock-adapter';
+import { writeFileSync } from 'fs';
+
+describe('Check axios-har-tracker', () => {
+
+    let axiosTracker, axiosMock, trackerMock, getHar
+    let fakeCallResponse = {
+      request: {
+        method: 'get',
+        url: 'http://fakeUrl',
+        httpVersion: 'HTTP/1.1',
+        cookies: [],
+        headers: [],
+        queryString: [],
+        headersSize: -1,
+        bodySize: -1
+      },
+      response: {
+        status: 200,
+        statusText: 'OK',
+        headers: [Array],
+        startedDateTime: '2020-12-16T10:53:02.000Z',
+        time: 140737012,
+        httpVersion: 'HTTP/1.1',
+        cookies: [],
+        bodySize: 13882,
+        redirectURL: '',
+        headersSize: -1,
+        content: [Object],
+        cache: {},
+        timings: [Object]
+      },
+      startedDateTime: '2020-12-16T10:53:02.034Z',
+      time: -1,
+      cache: {},
+      timings: {
+        blocked: -1,
+        dns: -1,
+        ssl: -1,
+        connect: -1,
+        send: 10,
+        wait: 10,
+        receive: 10,
+        _blocked_queueing: -1
+      }
+    };
+
+    beforeAll(async () => {
+      axiosTracker = new AxiosHarTracker(axios);
```

Comment:
Don't mix e2e test with test using mocks. Create two separate specs, one called `e2e.spec.ts` which will just call test endpoints from http://httpstat.us and and check if generated har file object is correct. Then if this will be working fine, you can create a separate spec (or few specs) for unit tests for each method in AxiosHarTracker class

Reply:
Create 4 tests cases
- request returns 200
- request returns 302
- request returns 404
- request returns 500

Reply:
that is a plan after the Holidays :)